packages/wallet/src/redux/protocols/application/reducer.ts

```diff
@@ -32,11 +37,22 @@ export function initialize(
   channelId: string,
   address: string,
   privateKey: string,
-  participants: ChannelParticipant[]
+  participants: ChannelParticipant[],
+  bytecode: string
 ): ProtocolStateWithSharedData<states.ApplicationState> {
   return {
-    protocolState: states.waitForFirstState({channelId, privateKey, address, participants}),
-    sharedData: registerChannelToMonitor(sharedData, APPLICATION_PROCESS_ID, channelId, [])
+    protocolState: states.waitForFirstState({
+      channelId,
+      privateKey,
+      address,
+      participants,
+      bytecode
+    }),
+    sharedData: registerBytecodeForChannel(
```

Comment:
I don't think we need to call `registerBytecodeForChannel` (or need that function?). Since `signAndInitialize`/`checkAndInitialize` accept a `byteCode` argument, they should initialize the `channelState` with the provided `byteCode`